lib/take2.rb

```diff
@@ -69,7 +69,7 @@ def call_api_with_retry(options = {})
       config.merge!(Take2.local_defaults(options)) unless options.empty?
       tries ||= config[:retries]
       begin
-        yield
+        yield(tries)
```

Comment:
@shved270189 why do you need it?

Reply:
@EvNomad in Bringg Client I want to log the tries count before call API:
```ruby
def some_post_request(url, post)
  with_retry  do |tries|
    Rails.logger("Bringg API #{url} call with #{params}. Try count: #{tries}")
  end
end
```

Reply:
@shved270189 I want to log **before** the first attempt?

Reply:
I am asking because you can log after each attempt with on_retry proc

Reply:
@EvNomad I want to log number of retry before to do something, not only before first attempt, I want to log before each attempt.

Reply:
> I am asking because you can log after each attempt with on_retry proc

Yes, I know, but I need to use some function params like url e.g. In on_retry proc I can not use method local variables.

Reply:
I still did not understand the use-case. Before the first attempt, you have the configured numbers of retries and then after each attempt is exactly the same as before.
After the first attempt, you have the same amount of retries as before the second attempt. So for ease of use you can do it with on_retry.

Reply:
@EvNomad yes, you are right, but in `on_retry  proc` I can not log method local variables.

Reply:
@EvNomad but I think, you are right. Let's check my feature PR before merge this one.

Reply:
> > I am asking because you can log after each attempt with on_retry proc
> 
> Yes, I know, but I need to use some function params like url e.g. In on_retry proc I can not use method local variables.

This is interesting. Maybe we could extend the proc arguments and pass there some hash.

Reply:
> > > I am asking because you can log after each attempt with on_retry proc
> > 
> > 
> > Yes, I know, but I need to use some function params like url e.g. In on_retry proc I can not use method local variables.
> 
> This is interesting. Maybe we could extend the proc arguments and pass there some hash.

Here: https://github.com/restaurant-cheetah/take2/blob/master/lib/take2.rb#L76

Reply:
@EvNomad Good idea. Thank you!